lib/tpm2_eventlog_yaml.c

```diff
@@ -250,101 +251,127 @@ static bool yaml_uefi_var(UEFI_VARIABLE_DATA *data, size_t size, UINT32 type) {
+                    if (start + slist->SignatureSize > size) {
+                        LOG_ERR("EventSize is inconsistent with actual data\n");
+                        break;
+                    }
+
+                    int signature_size = slist->SignatureListSize -
+                        sizeof(*slist) - slist->SignatureHeaderSize;
+                    if (signature_size < 0 || signature_size % slist->SignatureSize != 0) {
+                        LOG_ERR("Malformed EFI_SIGNATURE_LIST\n");
+                        break;
+                    }
+
+                    uint8_t *signature = (uint8_t *)slist +
+                        sizeof(*slist) + slist->SignatureHeaderSize;
+                    int signatures = signature_size / slist->SignatureSize;
+                    /* iterate through each EFI_SIGNATURE on the list */
+                    int i;
+                    for (i = 0; i < signatures; i++) {
+                        EFI_SIGNATURE_DATA *s = (EFI_SIGNATURE_DATA *)signature;
+                        char *sdata = calloc (1,
+                            BYTES_TO_HEX_STRING_SIZE(slist->SignatureSize-16));
+                        if (sdata == NULL) {
+                            LOG_ERR("Failled to allocate data: %s\n", strerror(errno));
+                            return false;
+                        }
+                        bytes_to_str(s->SignatureData, slist->SignatureSize-16,
+                            sdata, BYTES_TO_HEX_STRING_SIZE(slist->SignatureSize-16));
+                        uuid_unparse_lower(s->SignatureOwner, uuidstr);
+                        tpm2_tool_output("        - SignatureOwner: %s\n"
+                                         "          SignatureData: %s\n",
+                                         uuidstr, sdata);
+                        free(sdata);
+
+                        signature += slist->SignatureSize;
+                        start += slist->SignatureSize;
+                        if (start > size) {
+                            LOG_ERR("Malformed EFI_SIGNATURE_DATA\n");
+                            break;
+                        }
+                    }
+                    variable_data += slist->SignatureListSize;
+                }
+                return true;
             }
-
-            uint8_t *signature = (uint8_t *)slist +
-                sizeof(*slist) + slist->SignatureHeaderSize;
-            int signatures = signature_size / slist->SignatureSize;
-            /* iterate through each EFI_SIGNATURE on the list */
-            int i;
-            for (i = 0; i < signatures; i++) {
-                EFI_SIGNATURE_DATA *s = (EFI_SIGNATURE_DATA *)signature;
-                char *sdata = calloc (1,
-                    BYTES_TO_HEX_STRING_SIZE(slist->SignatureSize-16));
-                if (sdata == NULL) {
-                    LOG_ERR("Failled to allocate data: %s\n", strerror(errno));
+            else if ((strlen(ret) == 10 && strncmp(ret, "SecureBoot", 10) == 0)) {
```

Comment:
this and everywhere, start the else on same line as the previous if closing.

Reply:
yep, will fix